### PR TITLE
Use "optional_package" for scipy checks

### PR DESCRIPTION
**Pull request by bcipolli**
In a few places, scipy was still checked for in `try.. catch` blocks, rather than a call to `optional_package`. Switch those over.

Made another related changes as well, just because I was there:
* Turn `nibabel/tests/data/check_parrec_reslice.py` into an importable module, and fail more gracefully if `scipy` is not installed.

### CODE REVIEW

File: nibabel/tests/test_helpers.py [hunk 1]
```diff
@@ -42,15 +41,12 @@ def bz2_mio_error():
...
-        import scipy.io
-    except ImportError:
-        return True
     with InTemporaryDirectory():
         with Opener('test.mat.bz2', 'wb') as fobj:
             try:
+                import scipy.io
                 scipy.io.savemat(fobj, {'a': 1}, format='4')
-            except ValueError:
+            except ImportError, ValueError:
```
Comment (effigies) [inline nibabel/tests/test_helpers.py]: except (ImportError, ValueError):

File: nibabel/tests/test_helpers.py [hunk 2]
```diff
@@ -1,15 +1,14 @@
 """ Helper functions for tests
 """
+from io import BytesIO
 
 import numpy as np
+from nose.tools import assert_true
```
Comment (matthew-brett) [inline nibabel/tests/test_helpers.py]: I personally prefer test imports after other imports.

File: nibabel/tests/test_helpers.py [hunk 3]
```diff
@@ -42,15 +41,12 @@ def bz2_mio_error():
...
     commit 98ef522d99 (in scipy)
     """
-    try:
-        import scipy.io
-    except ImportError:
-        return True
     with InTemporaryDirectory():
         with Opener('test.mat.bz2', 'wb') as fobj:
             try:
+                import scipy.io
```
Comment (matthew-brett) [inline nibabel/tests/test_helpers.py]: On principle of isolating the line causing the error in the try block:

```
try:
    import scipy.io
except ImportError:
    return True
try:
    scipy.io.savemat(fobj, {'a': 1}, format='4')
except ValueError:
    return True
```

**Comment by matthew-brett** [comment]
Otherwise, looks good to me, thanks Ben.

File: nibabel/tests/test_helpers.py (hunk 3 above)
Comment (bcipolli) [inline nibabel/tests/test_helpers.py]: OK. In this case, I would just do the `optionalpackage` check. I thought the combined `try...catch` was more compact and easy to read, but glad to change.

File: nibabel/tests/test_helpers.py (hunk 2 above)
Comment (bcipolli) [inline nibabel/tests/test_helpers.py]: OK, glad to follow. Just to note, I was trying to follow the PEP8 convention for grouping imports into three: https://www.python.org/dev/peps/pep-0008/#imports . I'm not arguing for following that convention here, just to share where I was coming from.

**Comment by bcipolli** [comment]
Thanks @matthew-brett ; just made the requested changes (using `optional_package` instead of `try...catch` for scipy import check)

**Comment by matthew-brett** [comment]
Thanks for going through those changes.

Some travis test errors...

**Comment by bcipolli** [comment]
Fixed up & passing.

**Comment by matthew-brett** [comment]
Thanks for your patience ...

**Comment by bcipolli** [comment]
You too. Plus, this was way easier than last time! :)

**Pull request merged**